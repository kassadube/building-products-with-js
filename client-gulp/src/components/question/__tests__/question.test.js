import configureMockStore from 'redux-mock-store';
import  Question from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
};
const question = {
  id: '444',
  owner: user,
  text: 'Question text',
  answers: [{answer: 'Test answer'}],
};


test('# Question', () => {
  const onAnswer = () => {};
  const component = (<Question question={question} onAnswer={onAnswer} />);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
  // test events click
  const app = mount(component);
  const item = app.find('button');
  item.simulate('click');
});