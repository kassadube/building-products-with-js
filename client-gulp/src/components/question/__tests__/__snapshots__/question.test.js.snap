// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Question 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    Question text
    <div
      className="pull-right"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="profile/0"
      >
        test
         
      </Link>
       
    </div>
  </div>
  <div
    className="panel-body"
  >
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        Test answer
      </li>
    </ul>
  </div>
  <div
    className="panel-footer"
  >
    <form
      className="form-horizontal"
    >
      <div
        className="col-sm-10"
      >
        <input
          className="form-control"
          id="answerInput"
          placeholder="Enter your answer..."
          type="text"
        />
      </div>
      <button
        className="btn btn-default"
        onClick={[Function]}
        type="submit"
      >
        Answer
      </button>
    </form>
  </div>
</div>
`;
